.github/workflows/lint.yml

```
@@ -0,0 +1,71 @@
+name: Continuous integration code quality checks
+
+on:
+  pull_request:
+    branches: [trunk]
+  push:
+    branches: [trunk,add/linting-github-actions]
+
+jobs:
+  lint:
+    name: Parallel lint
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout repository
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # Setup PHP versions, run checks
+      - name: PHP setup
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 7.4
+      - name: Get composer cache directory
+        id: composer-cache
+        run: |
+          echo "::set-output name=dir::$(composer config cache-files-dir)"
+      - name: Cache composer dependencies
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+      - name: Install composer packages
+        run: composer install --no-progress
+
+      - name: Check for PHP errors
+        run: composer lint
+
+  phpcs_check:
+    name: PHPCS check
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout repository
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # Setup PHP versions, run checks
+      - name: PHP setup
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 7.4
+
+      - name: Get composer cache directory
+        id: composer-cache
+        run: |
+          echo "::set-output name=dir::$(composer config cache-files-dir)"
+      - name: Cache composer dependencies
+        uses: actions/cache@v1
```

```suggestion
        uses: actions/cache@c64c572235d810460d0d6876e9c705ad5002b353 # v2.1.6
```

Updates v1 to the latest v2 release https://github.com/actions/cache/releases/tag/v2.1.6